scipy/optimize/_linesearch.py

```diff
@@ -468,7 +469,8 @@ def extra_condition(alpha, phi):
         alpha_star = alpha1
         phi_star = phi_a1
         derphi_star = None
-        warn('The line search algorithm did not converge', LineSearchWarning)
+        warn('The line search algorithm did not converge',
+             LineSearchWarning, stacklevel=2)
```

Comment:
GH shows this like:

![image](https://github.com/scipy/scipy/assets/6570539/53d74434-ebe8-400c-b390-c30c01600976)

But shouldn't it be able to tell that that's not the relevant function? Do you think this is a bug?

Reply:
Yep definitely a bug, I noticed this in quite a few places. Might be worth browsing upstream but I'd imagine they are aware.